packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1599,9 +1645,14 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
         child: barrier,
       );
     }
-    barrier = IgnorePointer(
-      ignoring: animation!.status == AnimationStatus.reverse || // changedInternalState is called when animation.status updates
-                animation!.status == AnimationStatus.dismissed, // dismissed is possible when doing a manual pop gesture
+    barrier = AnimatedBuilder(
```

Comment:
is the full purpose of the animated builder is to rebuild the ignorepointer when _ignorePointer changes value? if so, you can call changedInternalState in the _maybeUpdateIgnorePointer to cause the rebuild instead

Reply:
Gotcha I can make this change—I imagine we'd then want to do the same around line 860 as well, right?

Curious—is this generally a better practice within the framework? My instinct is to think that AnimatedBuilder would perform better since it would scope the rebuilds to the part that's getting rebuilt, whereas `changedInternalState` would cause a rebuild of everything, no?

Reply:
So actually yeah when I add `changedInternalState` here, I start getting a failing test in `navigator_test` that `Routes don't rebuild just because their animations ended`—should I keep going with it and try to find a way to work around that?

Reply:
It looks like the test is there to make sure we don't rebuild the whole scope, it make sense to keep it as animated builder then.

Reply:
oh actually can you try _modalBarrier.markNeedsBuild